packages/@aws-cdk/aws-stepfunctions/README.md

```diff
@@ -258,6 +259,81 @@ const map = new stepfunctions.Map(this, 'Map State', {
 map.iterator(new stepfunctions.Pass(this, 'Pass State'));
 ```
 
+### Custom
+
+It's possible that the high-level constructs for the states or `stepfunctions-tasks` do not have
+the states or service integrations you are looking for. The primary reasons for this lack of
+functionality are:
+
+* A [service integration](https://docs.aws.amazon.com/step-functions/latest/dg/concepts-service-integrations.html) is available through Amazon States Langauge, but there are no Construct
+  classes for them
+* The state or state properties are available through Step Functions, but are not configurable
+  through constructs
+
+If a feature is not available, a `CustomState` can be used to supply any Amazon States Language
+JSON-based object as the state definition.
+
+[Code Snippets](https://docs.aws.amazon.com/step-functions/latest/dg/tutorial-code-snippet.html#tutorial-code-snippet-1) are available and can be plugged in as the state definition.
+
+All of the state properties will be embedded in the JSON that defines it. Custom states can be
+chained together and used like any other state. The `toStateJson` API returns the state JSON
+that was supplied when creating the custom state without any modification.
+
+Custom states can be chained together with any of the other states to create the state machine
+definition. You will also need to provide any permissions that are required to the `role` that
+the State Machine uses.
+
+The following example uses the `DynamoDB` service integration to insert data into a DynamoDB table.
+
+```ts
+import * as ddb from '@aws-cdk/aws-dynamodb';
+import * as cdk from '@aws-cdk/core';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+
+// create a table
+const table = new ddb.Table(this, 'montable', {
+  partitionKey: {
+    name: 'id',
+    type: ddb.AttributeType.STRING,
+  },
+});
+
+const finalStatus = new sfn.Pass(stack, 'final step');
+
+// States language JSON to put an item into DynamoDB
+// snippet generated from https://docs.aws.amazon.com/step-functions/latest/dg/tutorial-code-snippet.html#tutorial-code-snippet-1
+const stateJson = {
+  Type: 'Task',
+  Resource: 'arn:aws:states:::dynamodb:putItem',
+  Parameters: {
+    TableName: table.tableName,
+    Item: {
+      id: {
+        S: 'MyEntry',
+      },
+    },
+  },
+  ResultPath: null,
+  Next: finalStatus.id,
```

Comment:
The `Next` is not necessary, right?

Reply:
Will we honor the `null`? Do you have a test for that?

Reply:
it's not necessary, we do honor the `null`. I included it in the integ test I believe but happy to add it elsewhere

Reply:
By the way `null` is not going to work when used from Java/Python/C#. `null` and `undefined` are treated the same across the jsii interface.

This is actually a problem waiting to happen :(

Reply:
Also my comment about `Next`.